### PR TITLE
DCOS-19669: Add Mount for custom media type fields to FrameworkConfigurationForm

### PR DESCRIPTION
**Pull request by nLight**
To enable plugins and cleaner extensibility I've refactored the way we add custom media-type specific fields like YamlEditor and PlacementConstraints.

Closes DCOS-19669

### CODE REVIEW

File: tests/pages/catalog/packages/package/PackageTab-cy.js [hunk 1]
```diff
@@ -166,5 +166,81 @@ describe("Package Detail Tab", function() {
         .contains("Success")
         .should("to.have.length", 1);
     });
+
```
Comment (nLight) [inline tests/pages/catalog/packages/package/PackageTab-cy.js]: I copy/pasted Service's integration tests and they worked! 😊

**Comment by ahoskins** [comment]
Alfreddo test this please

File: src/js/components/PlacementConstraintsSchemaField.js [hunk 2]
```diff
@@ -60,34 +65,42 @@ export default class PlacementConstraintsFrameworkAdapter extends Component {
...
   }
 }
 
-PlacementConstraintsFrameworkAdapter.defaultProps = {
+PlacementConstraintsSchemaField.defaultProps = {
   onChange() {}
 };
 
-PlacementConstraintsFrameworkAdapter.propTypes = {
+PlacementConstraintsSchemaField.propTypes = {
```
Comment (ahoskins) [inline src/js/components/PlacementConstraintsSchemaField.js]: Change the propTypes to `fieldProps` of type object. And I think make it `.required` and remove the default props.

File: src/js/components/YamlEditorSchemaField.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+            showPrintMargin={false}
+            tabSize={2}
+            onChange={value => onChange(btoa(value))}
+          />
+        </div>
+        <FieldError>{errorMessage}</FieldError>
+      </div>
+    );
+  }
+}
```
Comment (ahoskins) [inline src/js/components/YamlEditorSchemaField.js]: Add propTypes?

File: src/js/components/SchemaField.js [hunk 4]
```diff
@@ -319,17 +263,28 @@ class SchemaField extends Component {
...
     const RADIO_SELECT_THRESHOLD = 4;
 
+    if (schema.media) {
+      return (
+        <MountService.Mount
+          type={`SchemaField:${schema.media.type}`}
+          schema={schema}
+          errorMessage={errorMessage}
+          autofocus={autofocus}
+          fieldProps={this.props}
```
Comment (ahoskins) [inline src/js/components/SchemaField.js]: Nice, I like the grouping of these props, makes it clear they're from the json-schema lib 👍

File: tests/_fixtures/cosmos/package-describe.json [hunk 5]
```diff
@@ -88,7 +88,7 @@
             },
             "uris": {
               "default": [
-
+
```
Comment (juliangieseke) [inline tests/_fixtures/cosmos/package-describe.json]: 🎨

**Comment by nLight** [comment]
Alfred please test this

**Comment by ahoskins** [comment]
Why is the hold merge label? Is this blocked by anything?

**Comment by nLight** [comment]
@ahoskins there's an issue with exact types we're using for placement. We're registering OSS and EE widgets with the same mount type, which results in duplication of widgets.

File: src/js/components/FrameworkConfigurationForm.js [hunk 6]
```diff
@@ -13,6 +15,22 @@ import UniversePackage from "#SRC/js/structs/UniversePackage";
 import CosmosErrorMessage from "#SRC/js/components/CosmosErrorMessage";
 import SchemaField from "#SRC/js/components/SchemaField";
 import StringUtil from "#SRC/js/utils/StringUtil";
+import PlacementConstraintsSchemaField
+  from "#SRC/js/components/PlacementConstraintsSchemaField";
+import YamlEditorSchemaField from "#SRC/js/components/YamlEditorSchemaField";
+
+MountService.MountService.registerComponent(
+  PlacementConstraintsSchemaField,
```
[the remark this answers is not in the document]
Comment (nLight) [reply src/js/components/FrameworkConfigurationForm.js]: After talking to Infinity we're not going to to that. They won't/can't have 2 versions of packages. So the plan is we render default version under those types and then _unregister_ all components from plugins and register new widgets.

[the remark this answers is not in the document]
Comment (nLight) [reply src/js/components/FrameworkConfigurationForm.js]: @orlandohohmeier please consider unblocking the pr

**Comment by juliangieseke** [comment]
Hm, the copied tests will fail after my PR is landed, I have put it on hold until this one is merged.

File: src/js/components/FrameworkConfigurationForm.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (orlandohohmeier) [reply src/js/components/FrameworkConfigurationForm.js]: Alright,  makes sense.

File: src/js/components/FrameworkConfigurationForm.js [hunk 7]
```diff
@@ -13,6 +15,22 @@ import UniversePackage from "#SRC/js/structs/UniversePackage";
...
+  "SchemaField:application/x-region-zone-constraints+json"
+);
+MountService.MountService.registerComponent(
+  PlacementConstraintsSchemaField,
+  "SchemaField:application/x-zone-constraints+json"
+);
+MountService.MountService.registerComponent(
+  YamlEditorSchemaField,
+  "SchemaField:application/x-yaml"
```
[the remark this answers is not in the document]
Comment (MatApple) [reply src/js/components/FrameworkConfigurationForm.js]: Makes sense!

**Pull request merged**